qcore/shared.py

```diff
@@ -1,109 +1,202 @@
 """
-Miscellaneous non-specific functions.
-Module which contains shared functions/values.
-
-@date 8 April 2016
+Miscellaneous functions whose usage is shared across a number of repositories.
 """
 
-from __future__ import print_function
-
+import re
 import subprocess
 import sys
-from io import IOBase
-
-# returns a list of stations
-# sample line in source file:
-#   171.74765   -43.90236 ADCS
-def get_stations(source_file, locations=False):
-    stations = []
-    station_lats = []
-    station_lons = []
-    with open(source_file, "r") as sp:
-        for line in sp.readlines():
-            if line[0] not in ["#", "%"]:
-                info = line.split()
-                stations.append(info[2])
-                if locations:
-                    station_lons.append(info[0])
-                    station_lats.append(info[1])
+from io import IOBase, StringIO
+from pathlib import Path
+from typing import AnyStr, Optional, Union
+
+import pandas as pd
+
+
+def get_stations(
+    station_ffp: Union[Path, str], locations: bool = False
+) -> Union[list[str], tuple[list[str], list[str], list[str]]]:
+    """Parse a station list file.
+
+    Sample line in source file:
+      171.74765   -43.90236 ADCS
+
+    Parameters
+    ----------
+    station_ffp : A Path or str
+        The file path of the station file.
+    locations : bool, default False
+        If True, also return locations.
+
+    Returns
+    -------
+    stations : list[str]
+        A list of station names in the file.
+    longitudes : list[float], if locations is True
+        The longitudes of the stations.
+    latitudes : list[float], if locations is True
+        The latitudes of the stations.
+    """
+    with open(station_ffp, "r", encoding="utf-8") as source_file_handle:
+        stations_text = source_file_handle.read()
+        # have to read and replace because pandas only allows one comment
```

Comment:
Do we actually have commented out lines using `%` in our station files? @sungeunbae @joelridden 
If not, then we should just change this to only support `#`?

Reply:
I don't recall ever seeing a % in these during my time. Probably safe to remove

Reply:
ok great I'll remove that. It will clean up the code.